onnxruntime/core/framework/session_state_utils.cc

```diff
@@ -386,6 +386,9 @@ common::Status SaveInitializedTensors(
     if (user_supplied_initializer_ids.find(entry.first) != user_supplied_initializer_ids.end()) {
       ort_value = *(session_options.initializers_to_share_map.at(name));
       LOGS(logger, INFO) << "Using user supplied initializer with name (" << name << ").";
+
+    } else if (graph.GetOrtValueInitializer(name, ort_value)) {
```

Comment:
The initializers stored in the graph are different in nature. Should we rename `GetOrtValueInitializer()` as `GetOrtValuePopulatedInitializer()`?

Reply:
Can you try out these changes? https://github.com/microsoft/onnxruntime/pull/24544

Reply:
> Can you try out these changes? #24544

It fixes DML EP issue. Thanks!